src/frontend/src/binder/relation/join.rs

```diff
@@ -91,11 +105,63 @@ impl Binder {
                     ))
                     .into());
                 }
-                bound_expr
+                (bound_expr, None)
             }
-            JoinConstraint::Using(_columns) => {
-                return Err(ErrorCode::NotImplemented("USING".into(), 1636.into()).into())
+            JoinConstraint::Using(columns) => {
+                let table_factor = table_factor.unwrap();
+                let right_table = get_table_name(&table_factor);
+                let mut columns_iter = columns.into_iter();
+                let first_column = columns_iter.next().unwrap();
+                let column_index = self.context.get_index(&first_column.value)?;
+                let mut left_table = self.context.columns[column_index].table_name.clone();
+                let mut binary_expr = Expr::BinaryOp {
+                    left: Box::new(Expr::CompoundIdentifier(vec![
+                        Ident::new(left_table.clone()),
+                        first_column.clone(),
+                    ])),
+                    op: BinaryOperator::Eq,
+                    right: Box::new(Expr::CompoundIdentifier(vec![
+                        right_table.clone().unwrap(),
+                        first_column,
+                    ])),
+                };
+                for column in columns_iter {
+                    left_table = self.context.columns[self.context.get_index(&column.value)?]
+                        .table_name
+                        .clone();
+                    binary_expr = Expr::BinaryOp {
+                        left: Box::new(binary_expr),
+                        op: BinaryOperator::Eq,
+                        right: Box::new(Expr::BinaryOp {
+                            left: Box::new(Expr::CompoundIdentifier(vec![
+                                Ident::new(left_table.clone()),
+                                column.clone(),
+                            ])),
+                            op: BinaryOperator::And,
+                            right: Box::new(Expr::CompoundIdentifier(vec![
+                                right_table.clone().unwrap(),
+                                column.clone(),
+                            ])),
+                        }),
+                    }
+                }
+                // We cannot move this into ret expression since it should be done before bind_expr
+                let relation = self.bind_table_factor(table_factor)?;
+                (self.bind_expr(binary_expr)?, Some(relation))
             }
         })
     }
 }
+
+fn get_table_name(table_factor: &TableFactor) -> Option<Ident> {
+    match table_factor {
+        TableFactor::Table { name, .. } => Some(name.0[0].clone()),
```

Comment:
As discussed offline, there may be no `table_name` after we support `NestedJoin`. It is also improper to use `name` when `alias` is present for `Table` (e.g. `explain select * from ab join bc as t using (b);`)

As the final BindContext refactor supporting all TableFactor syntax is more complicated than expected, we can merge this with limited support first.